executor/predictor/predictor_process.go

```diff
@@ -0,0 +1,131 @@
+package predictor
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/seldonio/seldon-core/executor/api/client"
+	"github.com/seldonio/seldon-core/executor/api/machinelearning/v1alpha2"
+	"github.com/seldonio/seldon-core/executor/api/payload"
+	"sync"
+)
+
+type PredictorProcess struct {
+	Client client.SeldonApiClient
+	Log    logr.Logger
+}
+
+func hasMethod(method v1alpha2.PredictiveUnitMethod, methods *[]v1alpha2.PredictiveUnitMethod) bool {
+	if methods != nil {
+		for _, m := range *methods {
+			if m == method {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (p *PredictorProcess) transformInput(node *v1alpha2.PredictiveUnit, msg payload.SeldonPayload) (payload.SeldonPayload, error) {
+	if (*node).Type != nil {
+		switch *node.Type {
+		case v1alpha2.MODEL:
```

Comment:
I am not sure if this is true or not, but looks like you can use a model as a transform_input?